### PR TITLE
Add more contraints to relationships test

### PR DESCRIPTION
**Pull request by me-ydv-5**

### CODE REVIEW

File: tests/tests_integration/test_api/test_relationships.py [hunk 1]
```diff
@@ -100,7 +103,7 @@ class TestRelationshipsAPI:
     def test_get_single(self, new_relationship):
         res = API_REL.retrieve(external_id=new_relationship.external_id)
         assert isinstance(res, Relationship)
-        assert res.dump()["confidence"] == 0.9
+        assert res.dump()["confidence"] == 1
```
Comment (me-ydv-5) [inline tests/tests_integration/test_api/test_relationships.py:106]: This was a failing edge case a few days ago in data viz team.

Comment (sanderland) [reply tests/tests_integration/test_api/test_relationships.py:106]: what was wrong? floating point roundoff ?

Comment (andreavs) [reply tests/tests_integration/test_api/test_relationships.py:106]: this edge case never failed for the retrieve endpoint - it would fail for the filter endpoint (for the curious, the confidence filter would run on all filter requests, and the default values were set to the extreme allowed values for confidence, and since the upper limit was also inclusive, every single filter request would run 0 <= confidence < 1 - big oopsie)

File: tests/tests_integration/test_api/test_relationships.py [hunk 2]
```diff
@@ -17,16 +17,19 @@
 
 
 @pytest.fixture
-def new_relationship():
+def new_relationship(new_label):
     external_id = uuid.uuid4().hex[0:20]
+    pre_existing_data_set = 3603480159997264
```
Comment (andreavs) [inline tests/tests_integration/test_api/test_relationships.py:22]: i guess this means the tests will fail if run on any other tenant than the sdk integration test tenant?

Comment (me-ydv-5) [reply tests/tests_integration/test_api/test_relationships.py:22]: yes. I couldn't think of any other way to by pass the mess of data-sets 😅

Comment (andreavs) [reply tests/tests_integration/test_api/test_relationships.py:22]: its good to test that data sets are working, but i would argue for either creating the data set with an external id, and then getting the internal id by fetching the data set, or setting the data set id by an env variable, although neither solution is very clean - how are data set tests handled elsewhere in the sdk?

Comment (me-ydv-5) [reply tests/tests_integration/test_api/test_relationships.py]: There is no test that includes data sets other than data set integration tests. Nevertheless I agree it should be fetched. So changed that part.

**Review by sanderland — APPROVED** [review]

File: tests/tests_integration/test_api/test_relationships.py (hunk 1 above)
Comment (me-ydv-5) [reply tests/tests_integration/test_api/test_relationships.py:111]: I've added the test for that in filter

**Review by andreavs — APPROVED** [review]
lgtm :+1:
